### PR TITLE
WA-4000 Disclosure consent SDK changes - PART 1

### PR DESCRIPTION
**Pull request by VindhyaRajagopal**
## The problem ##
WA-4000
[API Doc](https://developer.frollo.com.au/docs/frollo-api/b8dbb6df2728d-list-disclosure-consents)

## Checklist ##

Use the :white_check_mark: emoji for when it's been done; :no_entry: when it still needs work;
strikethrough when an item of the checklist does not apply to the changes.

* Is the destination branch correct & intended?:white_check_mark: 
* Is the code synced to latest (if applicable)?:white_check_mark: 
* Has the SDK example app build passed locally?:white_check_mark: 
* Is the correct JIRA ticket referenced?:white_check_mark: 
* Is the correct API docs/Confluence Page referenced?:white_check_mark: 
* Have all the AC's in the ticket passed?:white_check_mark: 
* Have you followed proper naming conventions in the code?:white_check_mark: 
* Has the code been properly covered with unit & android tests?:no_entry: 
* Have the full test suite (unit & android tests) passed locally?:no_entry: 
* Has the changelog.md, packages.md, overview.md, extras.md been updated?:no_entry: 
* Has the dokka task passed?:no_entry: 
* Have you committed database schema changes (if applicable)?:no_entry:

### CODE REVIEW

**Review by hailefrollo — APPROVED** [review]

File: src/main/kotlin/us/frollo/frollosdk/model/api/cdr/ExternalPartyResponse.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ import us.frollo.frollosdk.model.coredata.cdr.ExternalPartyType
 import us.frollo.frollosdk.model.coredata.cdr.SharingDuration
 import us.frollo.frollosdk.model.coredata.cdr.TrustedAdvisorType
 
-internal data class ExternalPartyResponse(
+data class ExternalPartyResponse(
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/api/cdr/ExternalPartyResponse.kt:27]: internal

File: src/main/kotlin/us/frollo/frollosdk/model/api/cdr/CDRDisclosureConsentsResponse.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package us.frollo.frollosdk.model.api.cdr
+
+import com.google.gson.annotations.SerializedName
+import us.frollo.frollosdk.model.coredata.cdr.CDRDisclosureConsentPermission
+import us.frollo.frollosdk.model.coredata.cdr.ConsentStatus
+
+internal data class CDRDisclosureConsentsResponse(
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/api/cdr/CDRDisclosureConsentsResponse.kt:7]: add the copy right header comments

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// implementing a database migration, as the column name would not change.
+
+@Entity(
+    tableName = "disclosure_consent",
+    indices = [
+        Index("id")
+    ]
+)
+
+data class CDRDisclosureConsent(
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:19]: rename as `DisclosureConsent `

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** The ID of the consent */
+    @PrimaryKey @ColumnInfo(name = "id") val consentId: Long,
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:22]: `name = "consent_id"`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.room.PrimaryKey
+import us.frollo.frollosdk.model.api.cdr.ExternalPartyResponse
+
+// Declaring the ColumnInfo allows for the renaming of variables without
+// implementing a database migration, as the column name would not change.
+
+@Entity(
+    tableName = "disclosure_consent",
+    indices = [
+        Index("id")
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:15]: `consent_id`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    ]
+)
+
+data class CDRDisclosureConsent(
+
+    /** The ID of the consent */
+    @PrimaryKey @ColumnInfo(name = "id") val consentId: Long,
+
+    /** The Status of disclosure consent */
+    @ColumnInfo(name = "status") val status: Long?,
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:25]: `ConsentStatus` and not nullable

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt (hunk 3 above)
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:19]: add documentation to the class

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** The Linked Collection Consent IDs (Optional) */
+    @ColumnInfo(name = "consent_ids") val consent_ids: List<Long>?,
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:28]: `linkedConsentIds`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** The Permissions given for the Disclosure Consent - same as Collection permissions (Optional) */
+    @ColumnInfo(name = "permissions") val permissions: List<CDRDisclosureConsentPermission>,
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:31]: `List<String>?`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** The disclosure duration in seconds of how long to disclose data for (Optional) */
+    @ColumnInfo(name = "disclosure_duration") val disclosureDuration: Int?,
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:34]: `Long?`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+    /** Stopped sharing at date. The date the consent expired or was withdrawn. (Optional)
+     *
+     * Date format for this field is ISO8601
+     * example 2011-12-03T10:15:30+01:00
+     */
+    @ColumnInfo(name = "sharing_stopped_at") val sharingStoppedAt: String?,
+
+    /**  External party (Optional) */
+    @ColumnInfo(name = "external_party") val externalParty: ExternalPartyResponse?
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsent.kt:51]: `: ExternalParty?`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsentPermission.kt [hunk 11]
```diff
@@ -0,0 +1,32 @@
+package us.frollo.frollosdk.model.coredata.cdr
+
+import androidx.annotation.Keep
+import com.google.gson.annotations.SerializedName
+import us.frollo.frollosdk.extensions.serializedName
+
+/** Indicates permissions given for disclosure consent */
+@Keep
+enum class CDRDisclosureConsentPermission {
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/CDRDisclosureConsentPermission.kt:9]: not needed. you can remove this enum

**Review by deepakmn86 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/us/frollo/frollosdk/model/api/cdr/DisclosureConsentsResponse.kt [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package us.frollo.frollosdk.model.api.cdr
+
+import com.google.gson.annotations.SerializedName
+import us.frollo.frollosdk.model.coredata.cdr.ConsentStatus
+
+internal data class DisclosureConsentsResponse(
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/api/cdr/DisclosureConsentsResponse.kt:22]: rename as `DisclosureConsentResponse`

File: src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/DisclosureConsent.kt [hunk 13]
```diff
@@ -0,0 +1,52 @@
+package us.frollo.frollosdk.model.coredata.cdr
```
Comment (deepakmn86) [inline src/main/kotlin/us/frollo/frollosdk/model/coredata/cdr/DisclosureConsent.kt:1]: add copy right header comments

**Pull request merged**